gammapy/modeling/models/spatial.py

```diff
@@ -1003,10 +1010,16 @@ def __init__(
         self, map, meta=None, normalize=True, interp_kwargs=None, filename=None,
     ):
         if (map.data < 0).any():
-            log.warning("Diffuse map has negative values. Check and fix this!")
+            log.warning("Map has negative values. Check and fix this!")
 
         if filename is not None:
             filename = str(make_path(filename))
+            if not os.path.isfile(filename):
```

Comment:
This check doesn't make any sense to me. Why would the file that is written to need to exist beforehand?

Reply:
My points was that `models.write` should write only the .yaml file and if users define **new maps** that their responsibility to save them  beforehand as it is an operation independent from the saving the model fit results.

 But ok I added a .write() line in to_dict if the file do not exit on disk, however if it already exits I just send a warning.
 I don't think we should overwrite by default or add an overwrite keyword in .to_dict() because then you likely need a different overwrite options for .fits and .yaml files in .write() as the latter may change more often.
 
I'm still convince that the best practice is to save new maps to disk before using them as templates and not rely on models.write to save templates...

Reply:
finally I added the two overwrite options for .fits and .yaml files in Models.write() ...